src/Microsoft.FeatureManagement/FeatureManager.cs

```diff
@@ -241,77 +241,108 @@ public ValueTask<Variant> GetVariantAsync(string feature, TargetingContext conte
                 throw new ArgumentNullException(nameof(context));
             }
 
-            return GetVariantAsync(feature, context, useContext: true, cancellationToken);
+            return GetVariantEvaluation(feature, context, useContext: true, cancellationToken);
         }
 
-        private async ValueTask<bool> IsEnabledWithVariantsAsync<TContext>(string feature, TContext appContext, bool useAppContext, CancellationToken cancellationToken)
+        private async ValueTask<Variant> GetVariantEvaluation(string feature, TargetingContext context, bool useContext, CancellationToken cancellationToken)
         {
-            bool isFeatureEnabled = false;
+            EvaluationEvent evaluationEvent = new EvaluationEvent
+            {
+                FeatureDefinition = await GetFeatureDefinition(feature).ConfigureAwait(false)
+            };
 
-            FeatureDefinition featureDefinition = await GetFeatureDefinition(feature).ConfigureAwait(false);
+            await EvaluateFeature(evaluationEvent, context, useContext, cancellationToken);
 
-            VariantDefinition variantDefinition = null;
+            return evaluationEvent.Variant;
+        }
 
-            if (featureDefinition != null)
+        private async ValueTask<bool> IsEnabledEvaluation<TContext>(string feature, TContext appContext, bool useAppContext, CancellationToken cancellationToken)
+        {
+            EvaluationEvent evaluationEvent = new EvaluationEvent
             {
-                isFeatureEnabled = await IsEnabledAsync(featureDefinition, appContext, useAppContext, cancellationToken).ConfigureAwait(false);
+                FeatureDefinition = await GetFeatureDefinition(feature).ConfigureAwait(false)
+            };
+
+            await EvaluateFeature(evaluationEvent, appContext, useAppContext, cancellationToken);
+
+            return evaluationEvent.IsEnabled;
+        }
+
+        private async Task<EvaluationEvent> EvaluateFeature<TContext>(EvaluationEvent evaluationEvent, TContext context, bool useContext, CancellationToken cancellationToken)
```

Comment:
evaluationEvent will be returned eventually. Would it be better to use `ref EvaluationEvent evaluationEvent`